### PR TITLE
DCM Output improvement, added feature to control preprocessing method

### PR DESCRIPTION
**Pull request by nickkprice**
Control preprocessing method with Resonator object initialization keyword Resonator(preprocessing_method="linear")
Initial logic for linear_normalization fix is included, but it only fixes a few cases right now
Shows 8 commits because my version control got messy but if you read through the changes it's fine

### CODE REVIEW

File: fit_resonator/README.md [hunk 1]
```diff
@@ -105,8 +105,11 @@ Must install pip module before installing pips for python 2. Once pip is install
 
 #### Section 1: Initial Setup
 Import code library with:
-`import fit_resonator.resonator as scres`
+`import fit_resonator.resonator as scres`:
```
[the remark this answers is not in the document]
Comment (crmcrae) [reply fit_resonator/README.md:108]: Is there an example user file with this set up in it as well?

File: requirements.txt [hunk 2]
```diff
@@ -1,8 +1,8 @@
 attr==0.3.*
 inflect==2.1.*
 lmfit==0.9.*
-matplotlib>=3.0.*
-numpy==1.16.*
+matplotlib>==3.5.3*
+numpy==1.21.*
```
Comment (crmcrae) [inline requirements.txt:5]: From issue #91:

matplotlib 3.5.3 has requirement numpy>=1.17, but you'll have numpy 1.16.6 which is incompatible.

Cna we address this here?

**Review by crmcrae — CHANGES_REQUESTED** [review]
A couple of questions/possible changes, otherwise looks awesome!

File: requirements.txt (hunk 2 above)
Comment (nickkprice) [reply requirements.txt:5]: Yes, it's probably best if I take a bit to just update the crucial packages to newer vesions and if anything breaks I'll address if before pull requesting again

File: fit_resonator/Sdata.py [hunk 3]
```diff
@@ -1300,10 +1317,20 @@ def preprocess_linear(xdata: np.ndarray, ydata: np.ndarray, normalize: int, outp
...
-    # normalize phase of S21 using linear fit
+
+    # Check for bed linear preprocessing outputs
+    # Redirect to circle preprocessing
     phase = np.unwrap(np.angle(ydata))
+    slope1, intercept1, r_value1, p_value1, std_err1 = stats.linregress(xdata, phase)
+
+    # Method for redirecting to circle, currently erroring data tends to have these attributes
+    if p_value1 < 1e-90 or std_err1 > 60:
+        return 'circle', None, None, None, None
```
[the remark this answers is not in the document]
Comment (nickkprice) [reply fit_resonator/Sdata.py:1328]: Yes. Should it have some other behavior if linear normalization fails?

**Pull request closed without merging**